lib/LightPanels/LightPanels.js

```diff
@@ -524,30 +548,22 @@ module.exports = kind(
 	* @private
 	*/
 	transitionFinished: function (sender, ev) {
-		if (ev.originator === this._currentPanel) {
-			if ((this._indexDirection < 0 && this.popOnBack && this.index < this.getPanels().length - 1) ||
-				(this._indexDirection > 0 && this.popOnForward && this.index > 0)) {
+		var panel = ev.originator;
+
+		if (panel === this._currentPanel) {
+			if ((this._indexDirection < 0 && (this.popOnBack || this.cacheViews) && this.index < this.getPanels().length - 1) ||
+				(this._indexDirection > 0 && (this.popOnForward || this.cacheViews) && this.index > 0)) {
 				this.popPanels(this.index, this._indexDirection);
 			}
-			if (this._currentPanel.postTransition) {
-				asyncMethod(this, function () {
-					this._currentPanel.postTransition();
-				});
-			}
-			if (this._garbagePanels && this._garbagePanels.length) {
-				this.finalizePurge();
-			}
+			if (this.popQueue && this.popQueue.length) this.finalizePurge();
+			this.transitioning = false;
+		}
 
-			if (this._currentPanel) {
-				this._currentPanel.removeClass('transitioning');
-				this._currentPanel.activated && this._currentPanel.activated();
-			}
+		panel.removeClass('transitioning');
 
-			if (this._previousPanel) {
-				this._previousPanel.removeClass('transitioning');
-				this._previousPanel.deactivated && this._previousPanel.deactivated();
-			}
-		}
+		asyncMethod(function () {
+			panel.postTransition();
```

Comment:
Noticed in the moonstone sample that this line can cause a runtime exception when a `transitionend` event bubbles to this control from something other than a panel (the ScrollThumb in the sample's case). We need to guard against that somehow here (and the prior line to remove the transitioning class). An `instanceof` check would probably work but it may be sufficient to move these into the `if` block above (assuming that makes logical sense here). An (better, i think) alternative would be to fire a custom event from LightPanel when `transitionend` fires and put the responsibility there. Thoughts?